components/autosize_textarea.tsx

```diff
@@ -0,0 +1,217 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {FormEvent, ChangeEvent, createRef, RefObject} from 'react';
+
+interface Props {
+    id?: string;
+    disabled?: boolean;
+    value?: string;
+    defaultValue?: string;
+    onChange?: (e: ChangeEvent<HTMLTextAreaElement>) => void;
+    onHeightChange?: (height: number, maxHeight: number) => void;
+    onInput?: (e: FormEvent<HTMLTextAreaElement>) => void;
+}
+
+interface DefaultProps {
```

Comment:
It doesn't look like you're using this interface anywhere, I assume you meant to declare it as part of the component?